keps/sig-storage/2848-transfer-volumesnapshots-namespaces/README.md

```diff
@@ -0,0 +1,795 @@
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+This KEP proposes a method for transferring VolumeSnapshot between namespaces. 
+
+Currently, VolumeSnapshot, and VolumeSnapshotContent are bound to specific namespace. The binding relationship between these two storage resources is one-to-one.
+This proposal targets allowing users to conveniently move these resources without manual creation
+of each independent object (VolumeSnapshot, VolumeSnapshotContent).
+
+Based on community design sessions, and inspired by [API for PVC Namespace Transfer](https://github.com/kubernetes/enhancements/pull/2110/files), the following is being proposed :-
+
+- `VolumeSnapshotTransferRequest` resources are created in the target namespace.
+- `VolumeSnapshotTransferApproval` resources are created in the source namespace.
+-  When a controller detects matching request/approval resources, a transfer in initiated.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+Allow Kubernetes users to quickly and conveniently transfer VolumeSnapshots across namespaces.
+### Goals
+- Use CRDs for VolumeSnapshot Namespace Transfer request/approval.
+- Edit external SnapshotController to accept/reject/init VolumeSnapshot transfer.
+- The source and target VolumeSnapshot will point to the same VolumeSnapshotContent.
+- Use Validation Webhook to reject any transfer contains secret.
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Transfer of resources other than VolumeSnapshot will not be discussed.
+- Transfer of VolumeSnapshots which contain secret.
+## Proposal
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+`VolumeSnapshotTransferRequest` resource:
```

Comment:
If we're planning to extend the same CRD to PVCs (which could be a good idea), then we would need to have at least a good design how PVCs will be transferred before going beta with snapshots, to make sure the API is correct. This is out of scope of this PR, but worth a note somewhere.

Reply:
@jsafrane I think holding on to have a good design which includes PVC has been discussed in the design sessions.
We concluded to move on with VolumeSnapshot without `secrets`, so that we can have something working and gather feedback. 

IMO we should move on with VolumeSnapshot only, and indeed before promoting to BETA we can revisit the design/impl to include PVCs, please correct me if I am wrong 🙏🏽

Reply:
@xing-yang @mkimuram +1, fixing it